src/meta/mod.rs

```diff
@@ -54,14 +54,17 @@ impl Meta {
             return Ok(None);
         }
 
-        if flags.display == Display::DirectoryItself {
+        if flags.display == Display::DirectoryItself && flags.layout != Layout::Tree {
             return Ok(None);
         }
 
         match self.file_type {
             FileType::Directory { .. } => (),
             FileType::SymLink { is_dir: true } => {
```

Comment:
Hey, I just realized that this might cause issues if we have something like below:
```
mkdir one
ln -s one one/two
```

This will go on a endless loop if we have dereference enabled. I am not really certain if we have to handle it though. I took a look at gnu ls and even thought it does not have `--tree` or `-R` it does not seem to handle the case were we symlink two itmes to each other and just error out with os error.
```
ln -s one two
ln -s two one
```

Do you have any other suggestion? If not, I think we can stick to this as of now hoping that users would be aware of `-L` use with `-R` or `--tree`. Also, when I tested it, it seemed to stop at few levels deep. I am not sure if this is an OS level restriction, but at least `lsd` does not go bonkers and eat up all the cpu if this happens.

Reply:
Just realized that gnu ls has `-R`.

```
/tmp/tree-test/one:
two
gls: /tmp/tree-test/one/two: not listing already-listed directory
```

This is the error that we get. Maybe we should do this check.

Reply:
Yes I think we should take from `ls` in general so ideally people could replace `ls` with `lsd` and behaviour is as expected maybe with some exceptions.

I will look into some way of "remembering" displayed files, and at how `ls` does it. Also, `tree` gives the following which I will try to recreate. Also perhaps we could add a "X directories, Y files" line too for `lsd --tree`

```bash
$ tree
.
└── one
    └── two -> ../one  [recursive, not followed]

2 directories, 0 files
```

Reply:
Btw, as of now it looks like `--tree` follows the symlink, but `tree` does not seem to follow symlinks by default. I think it would be better for us to do it that way too and only follow symlinks if there is `-L` with tree.

Reply:
"recursive" might not be the proper term. Like gnu ls, something like "already-listed" would be better as we probably need to make it so that all previously visited items are not listed, not just recursive (tracing if something is recursive might be unnecessary work).

Reply:
Hey, the changes to incorporate "remembering" visited paths, I guess, can be a separate  PR if that works for you. I can merge this in for now if you could add the rest of the tests as well.